### ISSUE TITLE
make pacui compatible with arch linux

### ISSUE THREAD

**Issue by excalibur1234** [excalibur1234/pacui#5]
there are multiple incompatibilities between manjaro and arch:

- which "reflector" command works everywhere on the world and gives the user the fastest available mirror/repo/server (even if it takes a long time to complete)?

- which "reflector" command works everywhere on the world and gives the user a fairly fast and available mirror/repo/server (this should not take a long time to complete)?

- do all the things to fix common manjaro issues also work in arch? (look here for reference: https://github.com/excalibur1234/pacui/blob/master/pacui#L1122 )

**Comment by thefallenrat** [comment]
I'll answer the second question

* Switch the `sudo pacman-mirrors -g -y` with maybe this : 

```
sudo reflector -f 5 --sort age && sleep 20 && sudo pacman -Syy
```
* * That `reflector` command will grab the fastest 5 mirrors and sort it with last server sync...


* You could try switch the `sudo pacman -Sy gnupg archlinux-keyring manjaro-keyring` into something like : 

```
sudo pacman -Sy gnupg $(pacman -Ssq '(-keyring)' | grep -v -i -E '(gnome|python)')
```

* * So that it will work for other distro that also arch-based.... 




* Also replace `sudo pacman-key --init && sudo pacman-key --populate archlinux manjaro` with : 

```
sudo pacman-key --init && sudo pacman-key --populate $(pacman -Qsq '(-keyring)' | grep -v -i -E '(gnome|python)' | sed s/-keyring// )
```
* The `sudo pacman -S --needed libsystemd systemd-sysvcompat systemd` are correct as that is recommended command to migrate back to native systemd arch as evidenced on last line in this site : 

http://systemd-free.org/migrate.php

> Revert to systemd
> If, perchance, you're not satisfied with the result you can always roll-back to systemd by executing the above two pacman commands inverted. 
> 
> ```
> # pacman -Rs sysvinit openrc eudev udev-openrc eudev-systemd dbus-openrc procps-ng-nosystemd syslog-ng-nosystemd udisks2-nosystemd consolekit polkit-consolekit upower-pm-utils udisks2-nosystemd desktop-privileges xorg-xwrapper acpid-openrc alsa-utils-openrc autofs-openrc consolekit consolekit-openrc cronie-openrc dbus-openrc cups-openrc displaymanager-openrc fuse-openrc haveged-openrc hdparm-openrc openssh-openrc samba-openrc syslog-ng-openrc avahi-openrc
> # pacman -S systemd libsystemd systemd-sysvcompat
> ```
> 
> 


And the rest are good enough for me!!

\*Note: If you are planning to commit sometime later, can I try to pull request? I need to know how to do it :3

**Comment by excalibur1234** [comment]
thanks for your suggested commands!

you can fork pacui to your github account.

then, do the necessary changes. please do not delete working code for manjaro. instead, put it in if-statements like
```
if [[ -e /usr/bin/pacman-mirrors ]]
then
sudo pacman-mirrors -g -y
fi

if [[ -e /usr/bin/reflector ]]
then
sudo reflector -f 5 --sort age && sleep 20 && sudo pacman -Syy
fi
```

when you have done all the necessary changes, click on "pull request" on your fork of pacui.

i have already noticed one mistake in your suggested code: you need to filter out one more -keyring package like:
`pacman -Slq '(-keyring)' | grep -v -i -E '(gnome|python|debian)'`

if you have trouble with the bash code, just tell me and i can do the necessary changes myself.

**Comment by excalibur1234** [comment]
should be fixed by c4bbc29cbf01205df04d4f1bffc3ee96e37f638b and 889b16b19364dc42a67d635e93c4d16b9250ca2d
i am still doing some testing before closing this.
